netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -607,7 +644,27 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
           logger.log(Level.WARNING, String.format("Received unexpected ping ack. "
               + "Expecting %d, got %d", p.payload(), ackPayload));
         }
-      } else {
+      }
+      else if (data.readLong() == 1234){     
+        pingreturn++;
+        int target = dataSinceLastPing * 2;
+        if(target > 8000000){
```

Comment:
Put it in a static final variable called `MAX_WINDOW` or similar to give it a semantic name. If you write it like `8 * 1024 * 1024` (or with 1000s) it is easier to read.